### PR TITLE
feat: Added ORC and PARQUET support among source tests

### LINKED ISSUE #1252: Support parquet and ORC files for Filesystem validations
Pandas supports [read_parquet()](https://pandas.pydata.org/pandas-docs/version/1.3/reference/api/pandas.read_parquet.html) and [read_orc()](https://pandas.pydata.org/docs/reference/api/pandas.read_orc.html). 

Add those as supported filetypes [here](https://github.com/GoogleCloudPlatform/professional-services-data-validator/blob/develop/data_validation/clients.py#L117) and in our documentation.

### PR DESCRIPTION
**Pull request by luispavaogoogle**
Closes #1252 

/gcbrun

### CODE REVIEW

File: docs/connections.md
Comment (helensilva14) [inline docs/connections.md:245]: Besides the section title, line 253 also needs to be updated to add the new file types

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
# Copyright 2024 Google LLC
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
        # Configuration Required Depending on Validation Type
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation JSON file.
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation ORC file.
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation CSV file.
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
    """Test schema validation Parquet file.
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

File: tests/system/data_sources/test_filesystem.py
Comment (helensilva14) [inline tests/system/data_sources/test_filesystem.py]:
```suggestion
```

**Review by helensilva14 — DISMISSED** [review]
Awesome work Luis! Just a few minor changes to do before I approve it

**Review by luispavaogoogle — COMMENTED** [review]
LGTM

File: tests/system/data_sources/test_filesystem.py
Comment (nj1973) [inline tests/system/data_sources/test_filesystem.py:41]: Nitpicky but constants should really be defined before and functions, i.e. these four constants should move above the `mock_get_connection_config` definition.

Approving anyway, I'll leave it up to you whether to move them or not.

**Review by nj1973 — APPROVED** [review]
LGTM aside from one nitpicky comment.

**Pull request merged**
